lib/fabrik/VirtualHostManager.js

```diff
@@ -0,0 +1,89 @@
+const _ = require('lodash');
+const Promise = require('bluebird');
+const bosh = require('../bosh');
+const utils = require('../utils');
+const VirtualHostInstance = require('./VirtualHostInstance');
+const BaseManager = require('./BaseManager');
+const VirtualHostAgent = require('./VirtualHostAgent');
+const logger = require('../logger');
+const errors = require('../errors');
+const BadRequest = errors.BadRequest;
+const ServiceBindingAlreadyExists = errors.ServiceBindingAlreadyExists;
+const ServiceBindingNotFound = errors.ServiceBindingNotFound;
+const ServiceInstanceNotFound = errors.ServiceInstanceNotFound;
+const NotFound = errors.NotFound;
+const mapper = require('./VirtualHostRelationMapper');
+
+class VirtualHostManager extends BaseManager {
+  constructor(plan) {
+    super(plan);
+    this.director = bosh.director;
+    this.agent = new VirtualHostAgent(this.settings.agent);
+    this.mapper = mapper.VirtualHostRelationMapper;
+  }
+
+  createVirtualHost(deploymentName, params, instanceId) {
+    logger.info(`Creating virtual host for deployment '${deploymentName}'...`);
+    return this.director.getDeploymentIps(deploymentName)
+      .then(ips => this.agent.createVirtualHost(ips, instanceId))
+      .then(() => this.mapper.createVirtualHostRelation(deploymentName, instanceId))
+      .tap(() => logger.info('+-> Created virtual host'));
+  }
+
+  deleteVirtualHost(deploymentName, instanceId) {
+    logger.info(`Deleting virtual host for deployment '${deploymentName}'...`);
+    return this.director.getDeploymentIps(deploymentName)
+      .then(ips => this.agent.deleteVirtualHost(ips, instanceId))
+      .then(() => this.mapper.deleteVirtualHostRelation(instanceId))
+      .tap(() => logger.info('+-> Deleted virtual host'))
+      .catchThrow(NotFound, new ServiceInstanceNotFound(instanceId));
+  }
+
+  createBinding(deploymentName, instanceId, binding) {
+    logger.info(`Creating binding '${binding.id}' for deployment '${deploymentName}', virtual host '${instanceId}'...`);
```

Comment:
merge both the log stmts into one. Else at times with many requests flowing in, these two might not be identified together.